src/armonik_cli/admin.py

```
@@ -43,73 +52,137 @@ def create_channel(arguments):
+    """
+    List sessions with filter options
+
+    Args:
+        client (ArmoniKSessions): ArmoniKSessions instance for session management
+        session_filter (Filter) : Filter for the session
+    """
+    page = 0
+    sessions = client.list_sessions(session_filter, page=page)
+    
+    while len(sessions[1]) > 0:
+        for session in sessions[1]:
+            print(f'Session ID: {session.session_id}')
+        page += 1
+        sessions = client.list_sessions(session_filter, page=page)
+
+    print(f'\nNumber of sessions: {sessions[0]}\n')
+
+
+
+def cancel_sessions(client: ArmoniKSessions, sessions: list):
+    """
+    Cancel sessions with a list of session IDs or all sessions running
+
+    Args:
+        client (ArmoniKSessions): Instance of the class with cancel_session method
+        sessions (list): List of session IDs to cancel
+    """
+    for session_id in sessions:
+        client.cancel_session(session_id)
+
+
+def list_tasks(client: ArmoniKTasks, session_ids: list, all: bool, creating: bool , error: bool):
+    """
+    List tasks associated with the specified sessions based on filter options
+
+    Args:
+        client (ArmoniKTasks): ArmoniKTasks instance for task management
+        session_ids (list): List of session IDs to filter tasks
+        all (bool): List all tasks regardless of status
+        creating (bool): List only tasks in creating status
+        error (bool): List only tasks in error status
+    """
+    for session_id in session_ids:
+
+        if all:
+            tasks_filter = TaskFieldFilter.SESSION_ID == session_id
+        elif creating:
+            tasks_filter = (TaskFieldFilter.SESSION_ID == session_id) & (TaskFieldFilter.STATUS == TASK_STATUS_CREATING)
+        elif error:
+            tasks_filter = (TaskFieldFilter.SESSION_ID == session_id) & (TaskFieldFilter.STATUS == TASK_STATUS_ERROR)
+        else:
+            print("SELECT ARGUMENT [--all | --creating | --error]")
+            return
```

should raise instead of return